spec/components/global_banner_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+describe "Global banner", type: :view do
+  def component_name
+    "global_banner"
+  end
+
+  it "renders nothing without any options" do
+    assert_empty render_component({})
+  end
+
+  it "renders with minimum required options" do
+    render_component(title: "Look here you", banner_version: 1)
+    assert_select ".gem-c-global-banner", text: "Look here you"
+  end
+
+  it "renders with a link on the title" do
+    render_component(title: "Look here you", banner_version: 1, title_href: "https://www.gov.uk")
+    assert_select ".gem-c-global-banner a.gem-c-global-banner__title[href='https://www.gov.uk']", text: "Look here you"
+  end
+
+  it "includes GA4 attributes when there is a link" do
+    render_component(title: "Look here you", banner_version: 1, title_href: "https://www.gov.uk")
+    assert_select 'a.gem-c-global-banner__title[data-ga4-link=\'{"event_name":"navigation","type":"global bar","section":"Look here you"}\']'
```

Comment:
Nitpick: should there be a test that checks for `data-module="ga4-link-tracker"` as well?

Reply:
Thanks, have added.